packages/mui-base/src/SwitchUnstyled/SwitchUnstyled.tsx

```diff
@@ -76,48 +73,45 @@ const SwitchUnstyled = React.forwardRef(function SwitchUnstyled(
   };
 
   const Root: React.ElementType = component ?? components.Root ?? 'span';
-  const rootComponentProps = resolveComponentProps(componentsProps.root, ownerState);
-  const rootProps: WithOptionalOwnerState<SwitchUnstyledRootSlotProps> = appendOwnerState(
-    Root,
-    {
-      ...otherProps,
-      ...rootComponentProps,
-      className: clsx(classes.root, stateClasses, className, rootComponentProps?.className),
+  const rootProps: WithOptionalOwnerState<SwitchUnstyledRootSlotProps> = useSlotProps({
+    elementType: Root,
+    externalSlotProps: componentsProps.root,
+    externalForwardedProps: otherProps,
```

Comment:
```suggestion
    externalForwardedProps: other,
```

Let's keep the name of the prop consistent. We would need to of course rename when destructuring the props too.